### PR TITLE
Form: adds prop to make red asterisk optional on required  fields

### PR DESCRIPTION
**Pull request by YamenSharaf**
Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

**Please note this does not break existing functionality as it is set to show asterisks by default.**

### CODE REVIEW

**Comment by YamenSharaf** [comment]
This is an added prop to Form component to address this [issue](https://github.com/ElemeFE/element/issues/12743)

https://github.com/ElemeFE/element/issues/12743

File: examples/docs/en-US/form.md [hunk 1]
```diff
@@ -838,6 +838,7 @@ All components in a Form inherit their `size` attribute from that Form. Similarl
 | label-position | position of label | string | left / right / top | right |
 | label-width | width of label, and all its direct child form items will inherit this value | string | — | — |
 | label-suffix | suffix of the label | string | — | — |
+| required-asterisk | whether required fields should have a red asterisk (star) beside their labels | boolean | — | true |
```
Comment (ziyoung) [inline examples/docs/en-US/form.md]: The new attribute can be named as `hide-required-asterisk` which default value is `false`.

Comment (YamenSharaf) [reply examples/docs/en-US/form.md]: Done!

**Comment by ziyoung** [comment]
@YamenSharaf Your branch has conflicts dev branch. A potential solution is rebasing header.

**Comment by YamenSharaf** [comment]
@ziyoung Done!

**Pull request merged**